scripts-components/toggler/src/index.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, EventListeners, DelegateEvent } from '@mangoweb/scripts-base'
+
+// data-target
+// data-class-name
+// data-toggle-others
+// data-prevent-default
+
+export default class Toggler extends Component<void> {
+	static componentName = 'Toggler'
+
+	getListeners = (): EventListeners => [['click', '.toggler', this.handleClick]]
+
+	handleClick(e: DelegateEvent<'click'>) {
+		const delegateTarget = e.delegateTarget as HTMLElement
+		const targetSelector = delegateTarget.dataset.target
+		const othersTargetSelector = delegateTarget.dataset.closeOthers
+		const toggleTarget = targetSelector ? document.querySelector(targetSelector) : delegateTarget
+		const closeOthersTargets = othersTargetSelector
+			? document.querySelectorAll(othersTargetSelector)
+			: null
+		const preventDefault = delegateTarget.dataset.preventDefault
+		const className = delegateTarget.dataset.className || 'is-active'
+		const tagContent = delegateTarget.dataset.tagContent || false
+
+		if (preventDefault) {
+			e.preventDefault()
+		}
+
+		if (closeOthersTargets) {
+			for (let i = 0; i < closeOthersTargets.length; i++) {
+				const target = closeOthersTargets[i];
+				if (target !== toggleTarget) {
+					target.classList.remove(className)
+				}
+			}
+		}
+
+		if (toggleTarget) {
+			toggleTarget.classList.toggle(className)
+		}
+
+		if (tagContent) {
+			const content = delegateTarget.innerText
+			delegateTarget.innerText = tagContent
+			delegateTarget.dataset.tagContent = content
```

Comment:
Moc se mi nelíbí ten nenápadný polymorfizmus toho `tag-content` atributu. Pokud tomu rozumím správně, tak to bude prohazovat hodnoty mezi tím, co je v tom atributu, a tím, co bylo uvnitř toho elementu původně, ale samotný význam je určený tím, jestli zrovna ten `delegateTarget` aktivní nebo ne. Ale jako asi ok…

Reply:
Myslím, že to byl nějaký project specific kus kód. Který Lukáš už odstranil.